osutil/disks/labels.go

```diff
@@ -48,93 +49,33 @@ func BlkIDEncodeLabel(in string) string {
-	stSlashEscapeX
-	stSlashEscapeXNum
-)
+var hexCode = regexp.MustCompile(`\\x[0-9a-f]{2}`)
 
 // BlkIDDecodeLabel decodes a string such as a filesystem or partition label
 // encoded by udev in BlkIDEncodeLabel for normal comparison, i.e.
 // "BIOS\x20Boot" becomes "BIOS Boot"
 func BlkIDDecodeLabel(in string) (string, error) {
-
-	const errFmtStr = "string is malformed, unexpected character '%c' not part of a valid escape sequence"
-
 	out := strings.Builder{}
-	escapedHexDigits := [2]rune{}
-	st := stNormal
-	for _, r := range in {
-		switch st {
-		case stNormal:
-			// check if this char is the beginning of an escape sequence
-			if r == '\\' {
-				st = stSlashEscape
-				continue
-			}
-			// otherwise write it
-			out.WriteRune(r)
-		case stSlashEscape:
-			// next char to check is 'x'
-			if r == 'x' {
-				st = stSlashEscapeX
-				continue
-			}
-			// otherwise it's a format error, "\" is not in the set of
-			// characters allowed, so if we see one that is not followed by an
-			// x, then the string is malformed and can't be decoded
-			return "", fmt.Errorf(errFmtStr, '\\')
-		case stSlashEscapeX:
-			// now we expect exactly two hex digits, since the encoding would
-			// have written valid multi-byte runes that are UTF8 directly
-			// without escaping, the only possible escaped runes are those which
-			// are one byte and not in the allowed set
-
-			// TODO: though can one have multi-byte runes that are not UTF8
-			// encodable? it seems the only possibilities are runes that are
-			// either in the surrogate range or that are larger than the maximum
-			// rune value - for now we will just ignore those
-
-			if strings.ContainsRune(`0123456789abcedf`, r) {
-				escapedHexDigits[0] = r
-				st = stSlashEscapeXNum
-				continue
-			}
-			return "", fmt.Errorf(errFmtStr, r)
-		case stSlashEscapeXNum:
-			// got one digit, make sure we get a second digit
-			if strings.ContainsRune(`0123456789abcedf`, r) {
-				escapedHexDigits[1] = r
-
-				// the escapedHexDigits can now be decoded and written out
-				v, err := strconv.ParseUint(string(escapedHexDigits[:]), 16, 8)
-				if err != nil {
-					// should be logically impossible, we ensured that only
-					// rune digits in the hexadecimal range above were put into this rune
-					// buffer
-					return "", fmt.Errorf("internal error, unable to parse escape sequence: %v", err)
-				}
-				escapedHexDigits = [2]rune{0, 0}
-				out.WriteRune(rune(v))
-				st = stNormal
-				continue
-			}
-			return "", fmt.Errorf(errFmtStr, r)
-		default:
-			return "", fmt.Errorf("internal error, unexpected parsing state")
+	pos := 0
+	for _, m := range hexCode.FindAllStringIndex(in, -1) {
+		start := m[0]
+		beforeMatch := in[pos:start]
+		if i := strings.IndexRune(beforeMatch, '\\'); i >= 0 {
+			return "", fmt.Errorf(`string is malformed, unparsable escape sequence at "%s"`, beforeMatch[i:])
 		}
+		out.WriteString(beforeMatch)
+		n, err := strconv.ParseUint(in[start+2:start+4], 16, 8)
```

Comment:
pretty sure this compiles down to the same instructions, but reads better:
```suggestion
		maybeHex := in[start+2:start+4]
		n, err := strconv.ParseUint(maybeHex, 16, 8)
```

Reply:
Thanks, I'll just call it `hex` because we are pretty sure it must be hexadecimal :-)

Reply:
Done.